src/linux/Packaging.Linux/install-from-source.sh

```diff
@@ -0,0 +1,101 @@
+#!/bin/sh
+
+SUDO_CMD=sudo
+
+# Parse script argument that allows users to customize sudo command
+for i in "$@"
+do
+    case "$i" in
+        --sudo-cmd=*)
+        SUDO_CMD="${i#*=}"
+        shift # past argument=value
+        ;;
+    esac
+done
+
+install_shared_packages() {
+    pkg_manager=$1
+    install_verb=$2
+
+    local shared_packages="git curl apt-transport-https"
+    for package in $shared_packages
+    do
+        if [ $pkg_manager = apk ]; then
+            $SUDO_CMD $pkg_manager $install_verb $package
+        else
+            $SUDO_CMD $pkg_manager $install_verb $package -y
+        fi
+    done
+}
+
+download_dotnet_script() {
+    curl -o dotnet-install.sh -L https://dot.net/v1/dotnet-install.sh
+    chmod +x ./dotnet-install.sh
+}
+
+distribution=$(sed -n '/^ID=/p' /etc/os-release | cut -c 4- | tr -d '"')
+case $distribution in
+    debian | ubuntu | linuxmint)
+        # add dotnet package repository/signing key
+        $SUDO_CMD apt update && $SUDO_CMD apt install wget -y
+        eval wget https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb -O packages-microsoft-prod.deb
```

Comment:
No need to `eval` this command.

Also, you used `curl` above, maybe we require only one of `curl` and `wget`? In other words, I would suggest to replace this call by:

```sh
curl -LO https://packages.microsoft.com/config/ubuntu/20.04/packages-microsoft-prod.deb
```

Also, do we have to use the `VERSION_ID` from `/etc/os-release` to avoid hard-coding `20.04` in this URL?

Reply:
> Also, you used curl above, maybe we require only one of curl and wget?

+1 using `curl` over `wget` (or visa versa).

> Also, do we have to use the VERSION_ID from /etc/os-release to avoid hard-coding 20.04 in this URL?

Does the .NET team publish runtimes for all the different distros we want/test with?

Reply:
Mmm this is a good point. Perhaps we should avoid using the debian package altogether and use `dotnet-install.sh` for Debian-based distros as well.

Reply:
Did some investigation here - using `dotnet-install.sh` for Debian-based distros is not a great idea. Apparently there are different package dependencies for different versions, and I don't want us to be in the business of maintaining that. So, I stuck to using the Debian package with the distribution/version in the repository URL.

This meant I had to split out `linuxmint` though, as there's no feed for it. I used the packages corresponding to the latest version here. Given that I don't _think_ many of our users are running linuxmint I feel like this is ok, but this gut feeling could be incorrect. Let me know if changes are recommended to support earlier versions.